logger_test.go

```diff
@@ -65,3 +65,27 @@ func TestLogger(t *testing.T) {
 		},
 	}, logs[4].Fields["Attributes"])
 }
+
+func TestLoggerPtrFie(t *testing.T) {
+	logger, exportLogs := logtest.Captured(t)
+	assert.NotNil(t, logger)
+	if globallogger.DevMode() {
+		logger = logger.With(otfields.AttributesNamespace)
+	}
+
+	str := "foo"
+	strPtr := &str
+	var nilStrPtr *string
+
+	logger.Info("ptr", log.Ptr("str", strPtr), log.Ptr("nilptr", nilStrPtr))
+
+	// This doesn't compile, meaning the implementation is correct.
+	// logger.Info("ptr", log.Ptr("str", nil))
```

Comment:
Sounds obvious, but until everyone has generics down, it felt better to double check that 🤣

Reply:
Something is only obvious if you look at it from the right perspective 😄